Examples/E2/Notification.cs

```diff
@@ -13,26 +13,43 @@ public class SmsNotification : INotification
     public void Send(string message) => Console.WriteLine($"SMS sent: {message}");
 }
 
-#region Thid block of code is not in your code base, it is a library(3rd party)
+
+#region This block of code is not in your code base, it is a library(3rd party)
 public class SlackNotification
 {
     public void PostMessageToChannel(string channel, string message)
     {
         Console.WriteLine($"Message sent to Slack channel {channel}: {message}");
     }
-} 
+}
 #endregion
+// We Should Create New Class And Inherit From SlackNotification And Impelement INotification InterFace
+// Then We Can Call PostMessageToChannel In INotification Send Method
+// For Channel Parameter We Can Get that from constructor or Hardcode That
+public class CustomSlackNotification : SlackNotification, INotification
+{
+    private readonly string _channel;
+
+    public CustomSlackNotification(string channel)
+    {
+        _channel = channel;
+    }
+    public void Send(string message)
+    {
+        PostMessageToChannel(_channel, message);
+    }
+}
 
 public class NotificationService()
 {
-    public ... Get(string type)
+    public INotification Get(string type)
     {
         if (type == "Email")
-            return new ...();
+            return new EmailNotification();
         else if (type == "SMS")
-            return new ...();
+            return new SmsNotification();
         else if (type == "Slack")
-            return new ...();
+            return new CustomSlackNotification("Channel1");
```

Comment:
if it is hard code, why you are passing it through the parameter. please use const in CustomSlackNotification class.

Reply:
I wanted to be free when using the class in other places .
And because this was an example, I decided to hardcode "Channel1" to the class